src/main/java/uk/gov/companieshouse/database/entity/ReasonEntity.java

```diff
@@ -11,15 +11,21 @@
 public class ReasonEntity implements Serializable {
 
     private final OtherReasonEntity other;
+    private final IllnessReasonEntity illnessReason;
 
-    public ReasonEntity(OtherReasonEntity other) {
+    public ReasonEntity(OtherReasonEntity other, IllnessReasonEntity illnessReason) {
         this.other = other;
+        this.illnessReason = illnessReason;
```

Comment:
This should be switched to use setters instead of a constructor because:

1. The number of constructor parameters will grow long if more reason types are added
2. The reasons are optional. This will end up with nulls being passed in

Reply:
Done